src/misc/msccl/msccl_setup.cc

```diff
@@ -260,7 +264,14 @@ ncclResult_t mscclSetupKernel(const void* sendBuff, void* recvBuff, size_t count
     ncclComm_t comm, hipStream_t stream) {
   mscclStatus& status = mscclGetStatus();
 
-  if (status.lastStream != stream && status.lastStream != nullptr) {
+  bool enableDoneEvent =
+#ifndef HIP_EVENT_DISABLE_FENCE
+	  (rcclParamMscclEnableDoneEvent() == 1);
+#else
+          true;
```

Comment:
Please correct me if my understanding is not correct, if HIP_ENABLE_DISABLE_EVENT is set, then enableDoneEvent should be set to false.

Reply:
With `HIP_EVENT_DISABLE_FENCE`, `doneEvent` will be instantiated with a bit hinting the hardware not to flush L2 cache, so we can safely set `enableDoneEvent` as `true`.